service/league_service/league_rater.py

```diff
@@ -18,4 +18,104 @@ def rate(
+            if player_div is None:  # Todo: refactor, because this is unreachable at the moment
+                cls._logger.warning("Doing placement again, because a division for id %s could not be found.",
+                                    current_score.division_id)
+            return cls._do_placement(league, current_score, rating)
+
+        score = cls._calculate_new_score(league, current_score, outcome, rating, player_div)
+        new_score, new_division_id = cls._calculate_division_change(league, score, current_score.division_id, player_div)
+        return LeagueScore(
+            new_division_id,
+            new_score,
+            current_score.game_count + 1
+        )
+
+    @classmethod
+    def _do_placement(cls, league, current_score, rating):
+        rating += config.RATING_MODIFIER_FOR_PLACEMENT
+        for div in league.divisions:
+            if div.max_rating >= rating >= div.min_rating:
+                new_score = div.highest_score * (rating - div.min_rating) / (div.max_rating - div.min_rating)
+                return LeagueScore(
+                    div.id,
+                    new_score,
+                    current_score.game_count + 1
+                )
+
+        highest_div = league.get_highest_division()
+        if rating > highest_div.max_rating:
+            return LeagueScore(
+                highest_div.id,
+                highest_div.highest_score,
+                current_score.game_count + 1
+            )
+
+        lowest_div = league.get_lowest_division()
+        if rating < lowest_div.min_rating:
+            return LeagueScore(
+                lowest_div.id,
+                0,
+                current_score.game_count + 1
+            )
+
+        cls._logger.warning("Could not find a suitable division in league %s for placement for rating %s",
+                            league, rating)
+        raise DivisionDoesNotExistError("Can't find suitable division for placement.")
+
+    @classmethod
+    def _calculate_new_score(cls, league, current_score, outcome, rating, player_div):
+        boost = 0
+        reduction = 0
+        higher_div = league.get_next_higher_division(player_div.id)
+
+        if rating > player_div.max_rating:
+            boost = config.POSITIVE_BOOST
+        elif rating < player_div.min_rating:
+            reduction = config.NEGATIVE_BOOST
+        # Boost for high rated players with low score to have players in top division sorted by rating
+        elif higher_div is None and current_score.score < player_div.highest_score \
+                * (rating - player_div.min_rating) / (player_div.max_rating - player_div.min_rating):
+            boost = config.HIGHEST_DIVISION_BOOST
+
+        new_score = current_score.score
+        if outcome is GameOutcome.VICTORY:
+            new_score += config.SCORE_GAIN + boost
+        elif outcome is GameOutcome.DEFEAT:
+            new_score -= config.SCORE_GAIN + reduction
```

Comment:
Maybe brackets make this easier to read?
```suggestion
            new_score -= (config.SCORE_GAIN + reduction)
```